src/ChunkSplitters.jl

```diff
@@ -74,51 +74,73 @@ is_chunkable(::Tuple) = true
 # Current chunks split types
 const split_types = (:batch, :scatter)
 
+# User defined constraint
+abstract type Constraint end
+struct FixedCount <: Constraint end
+struct FixedSize <: Constraint end
+
 # Structure that carries the chunks data
-struct Chunk{T}
+struct Chunk{T,C<:Constraint}
     itr::T
     n::Int
+    size::Int
     split::Symbol
 end
 is_chunkable(::Chunk) = true
 
 # Constructor for the chunks
-function chunks(data; n::Int, split::Symbol=:batch)
-    n >= 1 || throw(ArgumentError("n must be >= 1"))
+function chunks(data; n::Integer=0, size::Integer=0, split::Symbol=:batch)
```

Comment:
I don´t have strongly feelings about it, but when I started implementing this (very similarly, by the way), I used `nothing` as the default value of `size` and `n`. I don´t know if that can have some implication for compiler optimizations. If not, I'm not against leaving it as `zero`.

Reply:
Yeah, this is me being lazy I guess. We could make everything `Union{Nothing, Integer}`. But note that this would imply that we would need more type parameters for `Chunk` (or make some of the fields unions). I think I dislike this more than the `=0` "hack".

(If done right, I don't think it matters for the compiler.)